src/common/datasource/src/compression.rs

```diff
@@ -27,6 +27,7 @@ use tokio_util::io::{ReaderStream, StreamReader};
 
 use crate::error::{self, Error, Result};
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EnumIter, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
```

Comment:
I would use this for deser only `#[serde(rename_all(deserialize = "lowercase"))]` but would this create a problem when we serialize then deserialize from serialized result?

Reply:
Yes. Tried in the playground, `serde` can't deserialize the json string:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=63a3955bc5529951b97db199fdc6bc14